src/core/algorithms/ind/faida/inclusion_testing/hyperloglog.h

```diff
@@ -0,0 +1,235 @@
+#pragma once
+
+/**
+ * The content of this file is based on the HLL implementation made by
+ * Hideaki Ohno (https://github.com/hideo55). Modified by alexandrsmirn.
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <sstream>
+#include <stdexcept>
+#include <vector>
+
+#include "immintrin.h"
+
+#define HLL_HASH_SEED 313
+
+#if defined(__has_builtin) && (defined(__GNUC__) || defined(__clang__))
+
+#define _GET_CLZ(x, b) (uint8_t)std::min(b, ::__builtin_clz(x)) + 1
+
+#else
+
+inline uint8_t _get_leading_zero_count(uint32_t x, uint8_t b) {
+
+#if defined (_MSC_VER)
+    uint32_t leading_zero_len = 32;
+    ::_BitScanReverse(&leading_zero_len, x);
+    --leading_zero_len;
+    return std::min(b, (uint8_t)leading_zero_len);
+#else
+    uint8_t v = 1;
+    while (v <= b && !(x & 0x80000000)) {
+        v++;
+        x <<= 1;
+    }
+    return v;
+#endif
+
+}
+#define _GET_CLZ(x, b) _get_leading_zero_count(x, b)
+#endif /* defined(__GNUC__) */
+
+namespace hll {
+
+static const double pow_2_32 = 4294967296.0;       ///< 2^32
+static const double neg_pow_2_32 = -4294967296.0;  ///< -(2^32)
```

Comment:
Thanks for the good comment.
I don't know the desbordante's policy regarding what to do with code that was taken from other sources. It seems that such code is anyway modified to match the coding style in the desbordante (the same happened with conditional functional dependencies)
I'd like to hear opinion @polyntsov. It seems to me that it would be worth correcting the naming and, for example, correcting this remark.

Reply:
I wouldn't change the library code unless absolutely necessary. Or better yet, I wouldn't commit the third-party library code, instead adding it as a dependency.